core/gui/src/app/dashboard/component/user/user-workflow/user-workflow.component.ts

```diff
@@ -342,71 +342,84 @@ export class UserWorkflowComponent implements AfterViewInit {
     }
-    return false;
+
+    return upload$.pipe(
+      switchMap(() => from(this.search(true))),
+      tap(() => this.notificationService.success("Upload Successful")),
+      switchMap(() => of(false))
+    );
   };
 
   /**
    * process .zip file uploads
    */
-  private handleZipUploads(zipFile: Blob) {
+  private handleZipUploads(zipFile: Blob): Observable<any> {
     let zip = new JSZip();
-    zip.loadAsync(zipFile).then(zip => {
-      zip.forEach((relativePath, file) => {
-        file.async("blob").then(content => {
-          this.handleFileUploads(content, relativePath);
-        });
-      });
-    });
+    return from(zip.loadAsync(zipFile)).pipe(
+      switchMap(zip =>
+        from(
+          Promise.all(
+            Object.keys(zip.files).map(relativePath =>
+              zip.files[relativePath].async("blob").then(content => this.handleFileUploads(content, relativePath))
+            )
+          )
+        )
+      )
+    );
   }
 
   /**
    * Process .json file uploads
    */
-  private handleFileUploads(file: Blob, name: string) {
-    let reader = new FileReader();
-    reader.readAsText(file);
-    reader.onload = () => {
-      try {
-        const result = reader.result;
-        if (typeof result !== "string") {
-          throw new Error("Incorrect format: file is not a string");
-        }
-        const workflowContent = JSON.parse(result) as WorkflowContent;
-        const fileExtensionIndex = name.lastIndexOf(".");
-        let workflowName: string;
-        if (fileExtensionIndex === -1) {
-          workflowName = name;
-        } else {
-          workflowName = name.substring(0, fileExtensionIndex);
-        }
-        if (workflowName.trim() === "") {
-          workflowName = DEFAULT_WORKFLOW_NAME;
+  private handleFileUploads(file: Blob, name: string): Observable<any> {
+    return new Observable(observer => {
+      let reader = new FileReader();
+      reader.readAsText(file);
+      reader.onload = () => {
+        try {
+          const result = reader.result;
+          if (typeof result !== "string") {
+            throw new Error("Incorrect format: file is not a string");
+          }
+          const workflowContent = JSON.parse(result) as WorkflowContent;
+          const fileExtensionIndex = name.lastIndexOf(".");
+          let workflowName = fileExtensionIndex === -1 ? name : name.substring(0, fileExtensionIndex);
+          if (workflowName.trim() === "") {
+            workflowName = DEFAULT_WORKFLOW_NAME;
+          }
+          this.workflowPersistService
+            .createWorkflow(workflowContent, workflowName)
+            .pipe(untilDestroyed(this))
+            .subscribe({
+              next: uploadedWorkflow => {
+                this.searchResultsComponent.entries = [
+                  ...this.searchResultsComponent.entries,
+                  new DashboardEntry(uploadedWorkflow),
+                ];
+                observer.next();
```

Comment:
when success, use notification service to log a successful message

Reply:
If I add 'this.notificationService.success("Upload Workflow Successful");' here, there will be two pop-up notifications after success. Should I move the notification from above to below?

Reply:
> when success, use notification service to log a successful message

If I add 'this.notificationService.success("Upload Workflow Successful");' here, there will be two pop-up notifications after success. Should I move the notification from above to below?